config/config.example.yml

```diff
@@ -1,11 +1,147 @@
+# Invidious configuration file
+---
+# Number of threads to use for crawling videos from channels
 channel_threads: 1
+# Number of threads to use for refreshing subscription feeds
 feed_threads: 1
+
+# Log file path or STDOUT
+output: STDOUT
+
+# Log verbosity
+# All   = 0, Trace = 1
+# Debug = 2, Info  = 3
+# Warn  = 4, Error = 5
+# Fatal = 6, Off   = 7
+log_level: 3
+
+# Database configuration
 db:
   user: kemal
   password: kemal
   host: localhost
   port: 5432
   dbname: invidious
+
+# Use polling to keep decryption function up to date
+decrypt_polling: true
+
+# Crawling channel videos, threads should refresh
+# all videos uploaded by a channel
 full_refresh: false
+
+# Define if accessed via HTTPS
 https_only: false
-domain:
+
+# Signing key for CSRF tokens
+# randomly generated by default
+#hmac_key: 
+
+# External domain
+domain: replace.me
+
+# Use server-side notifications provided by YouTube
+# https://developers.google.com/youtube/v3/guides/push_notifications
+# Requires domain and hmac_key to be set
+use_pubsub_feeds: false
+
+# Show popular videos on this instance on the front page
+popular_enabled: true
+
+# Determine if CAPTCHA should be required for registration
+captcha_enabled: true
+
+# User login
+login_enabled: true
+
+# User registration
+registration_enabled: true
+
+# Instance statistics
+# Available on /api/v1/stats
+statistics_enabled: false
+
+# Instance admin account
+# account needs to exists
+admins: []
+
+# If running behind reverse proxy
+# an external port is required
+#external_port: 
+
+# Default preferences for
+# new and unregistered users
+default_user_preferences:
+  annotations: false
+  annotations_subscribed: false
+  autoplay: false
+  captions:
+  - ""
+  - ""
+  - ""
+  comments:
+  - youtube
+  - ""
+  continue: false
+  continue_autoplay: true
+  dark_mode: ""
+  latest_only: false
+  listen: false
+  local: false
+  locale: en-US
+  max_results: 40
+  notifications_only: false
+  player_style: invidious
+  quality: hd720
+  default_home: Popular
+  feed_menu:
+  - Popular
+  - Trending
+  - Subscriptions
+  - Playlists
+  related_videos: true
+  sort: published
+  speed: 1.0
+  thin_mode: false
+  unseen_only: false
+  video_loop: false
+  volume: 100
+
+# Disabled video IDs
+dmca_content: []
+
+# Automatically migrate database
+check_tables: false
+
+# Cache annotations from
+# Internet Archive
+cache_annotations: false
+
+# Optional banner to be displayed along top of page for announcements
+#banner:
+
+# Add HTTP Strict Transport Security header
+hsts: true
+
+# Disable proxying server-wide: options: 'dash', 'livestreams', 'downloads', 'local'
+disable_proxy: false
+
+# Force IPv4 / IPv6 connection to Google
+#force_resolve: 
+
+# HTTP port for internal webserver
+port: 3000
+
+# HTTP host for internal webserver
+host_binding: 0.0.0.0
+
+# Pool for HTTP connections to YouTube
+pool_size: 100
+
+# Cookies for connection to YouTUbe
+#cookies: 
+
+# Anti-Captcha configuration for reCaptcha
```

Comment:
Should mention that both anti-captcha, and captcha-monsters are available as provider.